packages/strapi-utils/lib/__tests__/sanitize-entity.test.js

```diff
@@ -0,0 +1,290 @@
+        { withPrivate: false, isOutput: true, includeFields: null },
+        {
+          ..._.pick(inputWithRelation, 'firstname', 'lastname'),
+          article: _.times(3, () => _.pick(article, ['name', 'content'])),
+        },
+      ],
+    ];
+
+    test.each(tests)(`Test n°%#`, (source, options, expected) => {
+      const { userRel: model } = models;
+      expect(sanitizeEntity(source, { ...options, model })).toStrictEqual(expected);
+    });
+  });
+
+  describe('With Dynamic Zone', () => {
+    test('Dynamic zone null', () => {
+      const { userDz: model } = models;
+      const dataSource = { ...inputWithDz, dz: null };
+
+      const expected = _.pick(dataSource, ['firstname', 'lastname', 'dz']);
+
+      expect(sanitizeEntity(dataSource, { model })).toStrictEqual(expected);
+    });
+
+    test('Dynamic zone with a basic component', () => {
+      const { userDz: model } = models;
+
+      const expected = {
+        ..._.pick(inputWithDz, ['firstname', 'lastname']),
+        dz: inputWithDz.dz.map(comp => _.pick(comp, ['__component', 'name', 'content'])),
+      };
+
+      expect(sanitizeEntity(inputWithDz, { model })).toStrictEqual(expected);
+    });
+  });
+
+  describe('Edge cases', () => {
+    test('It returns null if the model is nil', () => {
+      expect(sanitizeEntity(input, { model: null })).toBeNull();
+      expect(sanitizeEntity(input, { model: undefined })).toBeNull();
+    });
+
+    test(`It returns the input data as-is if it's not an object or an array`, () => {
+      const { user: model } = models;
+      expect(sanitizeEntity('foobar', { model })).toBe('foobar');
+      expect(sanitizeEntity(undefined, { model })).toBeUndefined();
+      expect(sanitizeEntity(null, { model })).toBeNull();
+    });
+
+    test('It sanitizes all entries if the input is an array', () => {
+      const dataSource = [_.clone(input), _.clone(input)];
+
+      const expected = [
+        _.pick(input, 'firstname', 'lastname'),
+        _.pick(input, 'firstname', 'lastname'),
+      ];
+
+      const { user: model } = models;
+
+      expect(sanitizeEntity(dataSource, { model })).toStrictEqual(expected);
+    });
+
+    test(`It should put the relation value as-is if it's nil'`, () => {
+      const dataSource = { ...inputWithRelation, article: null };
+      const expected = _.omit(dataSource, ['email', 'password']);
+
+      const { userRel: model } = models;
+
+      expect(sanitizeEntity(dataSource, { model })).toStrictEqual(expected);
+    });
+
+    test(`It should make sure that legacy data parsing returns an object`, () => {
+      const dataSource = {
+        toJSON() {
+          return input;
+        },
+      };
+
+      const { user: model } = models;
+
+      expect(sanitizeEntity(dataSource, { model })).toStrictEqual(
+        _.omit(input, ['email', 'password'])
+      );
+
+      expect(sanitizeEntity({ toJSON: () => null }, { model })).toBeNull();
```

Comment:
Why not use `jest.fn` and do `hasBeenCalled` ?

Reply:
Nice catch